app/src/main/java/ch/blinkenlights/android/vanilla/LibraryActivity.java

```diff
@@ -664,15 +713,39 @@ public boolean onCreateFancyMenu(Intent rowData, View view, float x, float y) {
 			if (type <= MediaUtils.TYPE_SONG || type == MediaUtils.TYPE_FILE)
 				fm.add(CTX_MENU_PLAY_ALL, 1, R.drawable.menu_play_all, R.string.play_all).setIntent(rowData);
 
+
+			if(type == MediaUtils.TYPE_ALBUM){
+				fm.add(CTX_MENU_PLAY_ALL_START_WITH_LAST_PLAYED, 1, R.drawable.menu_play_all, R.string.play_all_start_with_last_played).setIntent(rowData);
+			}
+
 			fm.add(CTX_MENU_ENQUEUE_AS_NEXT, 1, R.drawable.menu_enqueue_as_next, R.string.enqueue_as_next).setIntent(rowData);
 			fm.add(CTX_MENU_ENQUEUE, 1, R.drawable.menu_enqueue, R.string.enqueue).setIntent(rowData);
 
+
 			if (type == MediaUtils.TYPE_PLAYLIST) {
 				fm.add(CTX_MENU_RENAME_PLAYLIST, 0, R.drawable.menu_edit, R.string.rename).setIntent(rowData);
 			} else if (rowData.getBooleanExtra(LibraryAdapter.DATA_EXPANDABLE, false)) {
 				fm.add(CTX_MENU_EXPAND, 2, R.drawable.menu_expand, R.string.expand).setIntent(rowData);
 			}
 
+			int timestampstate = MediaLibrary.getAlbumUseSongTimestamp(this, rowData.getLongExtra("id", LibraryAdapter.INVALID_ID));
```

Comment:
This would block the UI thread: you shouldn't run SQL queries in the main thread.